setuptools/__init__.pyi

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Mapping, Optional, Sequence, Union
+
+import setuptools
+
+
+def setup(
+    name: str,
+    version: Optional[str] = None,
+    description: Optional[str] = None,
+    long_description: Optional[str] = None,
+    long_description_content_type: Optional[str] = None,
+    author: Optional[str] = None,
+    author_email: Optional[str] = None,
+    maintainer: Optional[str] = None,
+    maintainer_email: Optional[str] = None,
+    url: Optional[str] = None,
+    download_url: Optional[str] = None,
+    packages: Optional[Sequence[str]] = None,
```

Comment:
Some of these sequences allow newline- or comma- or space-separated strings. Some may be particular to tuples or lists. Probably specifying a sequence here is best though.

Reply:
Yes, some fields seemed a little convoluted - once we have mypy tests added to the test suite (and assuming there are tests for those invocations) we should see whether these type definitions are good enough.